packages/wonder-blocks-core/src/components/__tests__/unique-id-provider.test.js

```diff
@@ -112,26 +123,38 @@ describe("UniqueIDProvider", () => {
             expect(children).toHaveBeenCalledWith(SsrIDFactory);
         });
 
-        test("children calls after first get same unique id factory", () => {
+        test("children calls after first get same unique id factory", async () => {
             // Arrange
             const children = jest.fn(() => null);
             const nodes = (
                 <UniqueIDProvider mockOnFirstRender={true}>
                     {children}
                 </UniqueIDProvider>
             );
-            const wrapper = mount(nodes);
+            const {rerender} = render(nodes);
 
             // Act
-            wrapper.instance().forceUpdate();
+            rerender(nodes);
 
             // Assert
             // Check our forced render worked and we rendered three times.
-            expect(children).toHaveBeenCalledTimes(3);
-            // Check the second render gets a UniqueIDFactory instance.
-            expect(children.mock.calls[1][0]).toBeInstanceOf(UniqueIDFactory);
-            // Check the third render gets the same UniqueIDFactory instance.
-            expect(children.mock.calls[2][0]).toBe(children.mock.calls[1][0]);
+            waitFor(() => {
+                expect(children).toHaveBeenCalledTimes(3);
+            });
+
+            waitFor(() => {
+                // Check the second render gets a UniqueIDFactory instance.
+                expect(children.mock.calls[1][0]).toBeInstanceOf(
+                    UniqueIDFactory,
+                );
+            });
+
+            waitFor(() => {
+                // Check the third render gets the same UniqueIDFactory instance.
+                expect(children.mock.calls[2][0]).toBe(
+                    children.mock.calls[1][0],
+                );
+            });
```

Comment:
**question**: Shouldn't this be a single `waitFor` containing all the expectations? Also, doesn't `waitFor` need to be `await`'d?